x-pack/plugins/security_solution/public/management/components/console/components/command_input/components/command_input_history.tsx

```diff
@@ -0,0 +1,160 @@
+      };
+    });
+  }, [inputHistory]);
+
+  const selectableListProps: EuiSelectableProps['listProps'] = useMemo(() => {
+    return {
+      showIcons: false,
+      onKeyDownCapture: (ev) => {
+        // Works around an issue where the `enter` key event from the EuiSelectable gets capture
+        // by the outer `KeyCapture` input somehow. Likely due to the sequence of events between
+        // keyup, focus and the Focus trap component having the `returnFocus` on by default
+        if (ev.key === 'Enter') {
+          // @ts-expect-error
+          ev._CONSOLE_IGNORE_KEY = true;
+        }
+      },
+      windowProps: {
+        outerRef: optionListRef, // TODO:PT remove when https://github.com/elastic/eui/pull/5978 becomes available in kibana (task #4179)
+      },
+    };
+  }, []);
+
+  const renderSelectionContent: EuiSelectableProps['children'] = useCallback((list, search) => {
+    return list;
+  }, []);
+
+  const handleSelectableOnChange: EuiSelectableProps['onChange'] = useCallback(
+    (items: EuiSelectableOption[]) => {
+      optionWasSelected.current = true;
+
+      const selected = items.find((item) => item.checked === 'on');
+
+      dispatch({ type: 'updateInputPopoverState', payload: { show: undefined } });
+      dispatch({ type: 'updateInputPlaceholderState', payload: { placeholder: '' } });
+
+      if (selected) {
+        dispatch({ type: 'updateInputTextEnteredState', payload: { textEntered: selected.label } });
+      }
+
+      dispatch({ type: 'addFocusToKeyCapture' });
+    },
+    [dispatch]
+  );
+
+  const handleRenderOption = useCallback((option) => {
+    return <UserCommandInput input={option.label} />;
+  }, []);
+
+  // TODO:PT remove when https://github.com/elastic/eui/pull/5978 becomes available in kibana (task #4179)
+  const [checkObserverTrigger, setCheckObserverTrigger] = useState(1);
+
+  // TODO:PT remove when https://github.com/elastic/eui/pull/5978 becomes available in kibana (task #4179)
+  useEffect(() => {
+    let observer: MutationObserver;
+
+    if (optionListRef.current) {
+      observer = new MutationObserver((mutationList) => {
+        const focusedOption = mutationList.find((mutatedEl) =>
+          (mutatedEl.target as HTMLLIElement).classList.contains('euiSelectableListItem-isFocused')
+        );
+
+        if (focusedOption) {
+          dispatch({
+            type: 'updateInputPlaceholderState',
+            payload: {
+              placeholder: (focusedOption.target as HTMLLIElement).dataset.input as string,
+            },
+          });
+        }
+      });
+
+      observer.observe(optionListRef.current, {
+        attributes: true,
+        subtree: true,
+      });
+    } else {
+      setTimeout(() => {
+        if (isMounted) {
+          setCheckObserverTrigger(checkObserverTrigger + 1);
+        }
+      }, 5);
+    }
```

Comment:
note: This creates a new timeout infinitely when there are no commands in history. Since this particular `useEffect` block will be erased once we have the /pull/5978 in as you state in the TODO, this is okay for now. If not, then I'll suggest we create the timeout only once.

Reply:
Yeah, I'm also concerned about this but I think this is only while the EuiSelectable is being mounted. Did you tried it locally?

Reply:
Yeah I tried it locally and the set timeout fires infinitely when the selectable pop over is open when there are no commands in history.

Reply:
Then this is something we should take care about since it is a non desired scenario I guess

Reply:
Good catch! I'm going to test, but I see how this can be happening. I'll put in a check to ensure that we have history items and just exit if we don't. 

Also - good news. the change I need from Eui is in v60.0.0, which already has a [pending PR against kibana](https://github.com/elastic/kibana/pull/134919), so I should be able to remove all of this work-around code very soon